shell/platform/darwin/ios/framework/Source/FlutterView.mm

```diff
@@ -16,8 +16,16 @@
 #import "flutter/shell/platform/darwin/ios/ios_surface_software.h"
 #include "third_party/skia/include/utils/mac/SkCGUtils.h"
 
+static BOOL IsWideGamutSupported() {
+  // This predicates the decision on the capabilities of the iOS device's
+  // display.  This means external displays will not support wide gamut if the
+  // device's display doesn't support it.  It practice that should be never.
+  return UIScreen.mainScreen.traitCollection.displayGamut != UIDisplayGamutSRGB;
```

Comment:
What will happen when moving the window to a monitor that doesn't support wide gamut?

Reply:
I believe the extended range values will be clamped in the driver.  So it is technically wasteful if your display doesn't support wide gamut but it doesn't require handling operating system callbacks about transitioning surfaces from one display to another.  We don't have a final cost value for the wide gamut support since I'm still working on optimizations, but I think this is the right tradeoff with the information we have.

Reply:
Yea, I have to almost believe the display driver has to clamp the values but this is just speculation.  We should test it out.  We'll never be able to have an integration test for this unfortunately.  If it doesn't work this isn't the appropriate PR to address it.

Filed an issued: https://github.com/flutter/flutter/issues/120485